src/main/java/com/BlogSite/TestBlogProject/repositories/PostRepository.java

```diff
@@ -10,5 +10,10 @@
 
 @Repository
 public interface PostRepository extends JpaRepository<Post, Long> {
+    @Query("select p from Post p " +
+            "JOIN FETCH p.user u " +
+            "where u.username = :username")
+    List<Post> findByUserUsername(@Param("username") String username);
```

Comment:
i think this method should work without query. And when you will set FetchType=EAGER i think it will solve your issue with 2 requests and it will be possible to remove custom query